op_sparsemax.go

```diff
@@ -94,7 +116,7 @@ func (op *sparsemaxOp) Do(inputs ...Value) (Value, error) {
 		return nil, fmt.Errorf("invalid input type for Sparsemax, expected float64 or float32, got: %v", inputTensor.Dtype())
 	}
 
-	return tensor.New(tensor.Of(inputTensor.Dtype()), tensor.WithShape(inputTensor.Size()), tensor.WithEngine(inputTensor.Engine()), tensor.WithBacking(output)), nil
+	return tensor.New(tensor.Of(inputTensor.Dtype()), tensor.WithShape(inputTensor.Shape()...), tensor.WithEngine(inputTensor.Engine()), tensor.WithBacking(output)), nil
```

Comment:
`.Shape().Clone()...` - this will prevent funny shared shape things from happening

Reply:
fixed